### PR TITLE
Updated installation instructions

### PR DESCRIPTION
**Pull request by LisaFC**
Added per-language installation instructions (either one-liners where available or links to installation docs) for gRPC and protoc to the overview. Am missing some details for protocol buffers in Python and Ruby....

### CODE REVIEW

**Comment by LisaFC** [comment]
@hsaliak @stanley-cheung @ejona86 @nathanielmanistaatgoogle @tbetbetbe @jtattermusch @ctiller @murgatroid99 @iamqizhao

**Comment by LisaFC** [comment]
If you're on the corp network, I have a preview running on my machine: http://lcarey0-l.dub.corp.google.com:4000/docs/#install-grpc

File: docs/index.md [hunk 1]
```diff
@@ -251,6 +248,131 @@ $ git clone https://github.com/grpc/grpc.git
...
+    <li><a href="#objective-c_install">Objective-C</a></li>
+    <li><a href="#php_install">PHP</a></li>
+  </ul>
+<div id="java_install">
+This example is also part of Java gRPC itself's build, so running it is a little different to creating and running a regular project. To install and build it follow the instructions in the [Quickstart](https://github.com/grpc/grpc-java/tree/master/examples).
+
+To install the runtime for your own projects, add the following to your build files:
+
+Gradle:
+<pre>compile 'io.grpc:grpc-all:0.13.1'</pre>
```
Comment (ejona86) [inline docs/index.md]: So grpc-java will need to bump this documentation with each release. I'm a bit concerned that it's going to bitrot, but we can try it out for a while. It does seem Java is the only language that will have a problem; I can't find a good link to mimic what C# is doing (all the links I have that are useful include the version number).

Comment (LisaFC) [inline docs/index.md]: Hmm, yes. We do have a site-wide doc variable for the release part of various Github URLs (currently set to release-0_13) but that's not exactly what we want here. I was going to suggest trying to add some Javascript to set some text from that but the URL variable doesn't contain enough information (it has the 0 and the .13 but not the .1).

**Comment by LisaFC** [comment]
Also @hsaliak, presume we want to publish this along with the "improved installation!" blog post?

**Comment by LisaFC** [comment]
Anyone else got any input on this? I'd like to have this info updated on the site along with the announcement/blog post about improved installation process.

**Comment by ejona86** [comment]
The java changes LGTM.

File: docs/index.md [hunk 2]
```diff
@@ -251,6 +248,131 @@ $ git clone https://github.com/grpc/grpc.git
...
+<div id="ruby_protoc">
+First <a href="https://github.com/google/protobuf/blob/master/README.md">install <code>protoc</code></a> from its Github repository.
+
+<!--Where do they get the gRPC protoc plugin? Also what's the command to install the protobuf gem, and do they need that too?-->
+</div>
+<div id="node_protoc">
+The Node.js gRPC library currently dynamically generates the necessary gRPC code at runtime, so you don't need to install a protocol buffer compiler yourself.
+</div>
+<div id="csharp_protoc">
+On Mac, Linux, and Windows, install the <a href="https://www.nuget.org/packages/Google.Protobuf">Google.Protobuf</a> and <a href="https://www.nuget.org/packages/Grpc.Tools">gRPC.Tools</a> NuGet packages from your IDE (Visual Studio, Monodevelop, Xamin Studio).
```
Comment (jtattermusch) [inline docs/index.md]: Xamin -> Xamarin

File: docs/index.md [hunk 3]
```diff
@@ -251,6 +248,131 @@ $ git clone https://github.com/grpc/grpc.git
...
+On Mac, Linux, and Windows, run:
+<pre>gem install grpc
+</pre>
+</div>
+<div id="node_install">
+On Mac, Linux, and Windows, run:
+<pre>npm install grpc</pre>
+</div>
+<div id="csharp_install">
+On Mac, Linux, and Windows, install the [gRPC NuGet package](https://www.nuget.org/packages/Grpc/) from your IDE (Visual Studio, Monodevelop, Xamin Studio).
```
Comment (jtattermusch) [inline docs/index.md]: Xamin -> Xamarin

Comment (LisaFC) [inline docs/index.md]: Done.

File: docs/index.md (hunk 2 above)
Comment (LisaFC) [inline docs/index.md]: Done.

Comment (jtattermusch) [inline docs/index.md]: for C# we only ship precompiled   protoc and grpc_csharp_plugin  for Windows platform.
For Linux and Mac, protoc and grpc_csharp_plugin still need to be installed using Homebrew.  

Also, this is changing soon and we will be shipping precompiled  protoc and grpc_csharp_plugin for all of Windows, Mac and Linux (possibly in the next release).

Comment (LisaFC) [inline docs/index.md]: Ok, done. We can update in the next release.

**Comment by hsaliak** [comment]
This looks good to me. Thanks for making the changes.

**Comment by jtattermusch** [comment]
please add "ready to merge" label if you want me to merge this.

**Comment by LisaFC** [comment]
It still needs some Python and Ruby info for protoc & plugins installation, once I have that it's good to go.

**Comment by LisaFC** [comment]
Python and Ruby people? @hsaliak, can you nag, er, I mean, persuade someone? I'd like to get this published.

**Comment by soltanmm** [comment]
Python looks fine; the protoc plugin thing'll be available Soon™.

**Comment by LisaFC** [comment]
@soltanmm - Thanks - though what do they do for the plugin before "soon"? :) At the moment I'm only telling them to get protoc itself from the repo, they'll presumably need the gRPC plugin as well to actually generate code....

**Comment by murgatroid99** [comment]
The Ruby instructions look good to me.

**Comment by soltanmm** [comment]
Right, so... they have to acquire it from source at the moment. -.-'

`git clone https://github.com/grpc/grpc.git $REPO`
`cd $REPO`
`make grpc_python_plugin`
`cp bins/opt/grpc_python_plugin $SOMEWHERE_ON_PATH`

**Comment by LisaFC** [comment]
@murgatroid99 - as per Python, though, where do they get the gRPC protoc plugin? Build from source as well?

**Comment by hsaliak** [comment]
Yes ideally the protoc binary can be obtained from the latest protobuf release 
https://github.com/google/protobuf/releases and the protoc plugin (for ruby and python) need to be compiled from source.

**Comment by LisaFC** [comment]
Ok, have Ruby instructions now, is everyone else happy?

**Comment by nathanielmanistaatgoogle** [comment]
I'm satisfied.

**Comment by murgatroid99** [comment]
LTGM

**Pull request merged**